lib/wasix/src/http/web_http_client.rs

```diff
@@ -0,0 +1,272 @@
+use anyhow::{Context, Error};
+use futures::future::BoxFuture;
+use http::header::{HeaderMap, HeaderValue, IntoHeaderName};
+use wasm_bindgen::{JsCast, JsValue};
+use wasm_bindgen_futures::JsFuture;
+use web_sys::{RequestInit, RequestMode, Window, WorkerGlobalScope};
+
+use crate::{
+    http::{HttpClient, HttpRequest, HttpRequestOptions, HttpResponse},
+    utils::web::js_error,
+};
+
+#[derive(Debug, Default, Clone, PartialEq)]
+#[non_exhaustive]
+pub struct WebHttpClient {
+    default_headers: HeaderMap,
+}
+
+impl WebHttpClient {
+    pub fn new() -> Self {
+        WebHttpClient {
+            default_headers: HeaderMap::new(),
+        }
+    }
+
+    pub fn with_default_header(
+        &mut self,
+        name: impl IntoHeaderName,
+        value: HeaderValue,
+    ) -> &mut Self {
+        self.default_headers.insert(name, value);
+        self
+    }
+
+    pub fn extend_default_headers(&mut self, map: HeaderMap) -> &mut Self {
+        self.default_headers.extend(map);
+        self
+    }
+}
+
+impl HttpClient for WebHttpClient {
+    fn request(&self, mut request: HttpRequest) -> BoxFuture<'_, Result<HttpResponse, Error>> {
+        let (sender, receiver) = futures::channel::oneshot::channel();
+
```

Comment:
Note that `reqwest` actually supports the browser. 
Any drawbacks to just using reqwest for js builds as well?

(I think one potential complication is that the futures used by reqwest in browser builds aren't send/sync...)

Reply:
The biggest benefit I see is being able to drop `reqwest` and all the dependencies it pulls in. 

That'll have massive benefits in terms of both compile times and the size of the `*.wasm` file every user's browser needs to download.

Reply:
That's a very valid argument.